servlet/src/main/java/io/undertow/servlet/handlers/ServletPathMatchesData.java

```diff
@@ -84,14 +84,10 @@ public ServletPathMatch getServletHandlerByPath(final String path) {
     }
 
     private ServletPathMatch handleMatch(final String path, final PathMatch match, final int extensionPos) {
-        if (match.extensionMatches.isEmpty()) {
+        if (extensionPos == -1 || match.extensionMatches.isEmpty()) {
             return new ServletPathMatch(match.defaultHandler, path, match.requireWelcomeFileMatch);
         }
-        if (extensionPos == -1) {
-            return new ServletPathMatch(match.defaultHandler, path, match.requireWelcomeFileMatch);
-        }
-        final String ext;
-        ext = path.substring(extensionPos + 1, path.length());
+        final String ext = path.substring(extensionPos + 1);
```

Comment:
No functional change here, I removed the unnecessary second argument to substring, and combined the two conditionals with the same result